### PR TITLE
replace archy with a specific implementation, refactor timetree

### PR DESCRIPTION
**Pull request by Uzlopak**
archy is 9 years old and you can not find the repo on github anymore. 
this PR replaces the need for archy and removes some double processing

added also a unit test as #205 got also fixed, i presume. 

Edit:
removed unreachable code and added jsdoc types. 
test coverage 100% for time-tree.js

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
it might be worth moving that archy clone in another repo, I think we use it in find-my-way as well

**Comment by Uzlopak** [comment]
@mcollina 
I checked it and find-my-way does not use archy. Also this is an adaptation for the specific need of avvio.

@eomm

Can you have a look regarding bug #205?

File: test/gh-issues/bug-205.test.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+const { test } = require('tap')
+const boot = require('../..')
+
+test('should print the time tree', (t) => {
+  t.plan(2)
+  const app = boot()
+
+  app.use(function first (instance, opts, cb) {
+    const out = instance.prettyPrint().split('\n')
+    t.equal(out[0], 'bound root undefined ms')
```
Comment (Eomm) [inline test/gh-issues/bug-205.test.js:12]: undefined?

Comment (Uzlopak) [reply test/gh-issues/bug-205.test.js:12]: yes, what do you prefer? 0 is misleading.  at that point there is no diff. So what should be printed?

**Comment by Eomm** [comment]
> it might be worth moving that archy clone in another repo, I think we use it in find-my-way as well

I have a fork - https://github.com/Eomm/node-archy
I may publish it with a bit rework

File: test/gh-issues/bug-205.test.js (hunk 1 above)
Comment (Eomm) [reply test/gh-issues/bug-205.test.js:12]: For something that says "not loaded yet", Maybe `-1`?

Comment (Uzlopak) [reply test/gh-issues/bug-205.test.js:12]: k, will do. Currently writing unit tests.

**Comment by Uzlopak** [comment]
I removed unreachable code, added jsdoc typings

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**